### PR TITLE
Prove that classical reals implement constructive reals. 

### PR DESCRIPTION
**Pull request by VincentSe**
**Kind:** feature.

This proof allows to apply results from constructive reals to classical reals. Also, move constructive `CRsum`, `CRmin`, `CRmax` and their theorems to CoRN.

In stdlib, only leave what is necessary to prove that all inhabitants of `ConstructiveReals` are equivalent, and that `R` is an instance of `ConstructiveReals`.

### CODE REVIEW

**Comment by VincentSe** [comment]
@MSoegtropIMC As you can see from the comment of @Zimmi48 there https://github.com/coq/coq/pull/12287, it is too soon to develop min, max and powers in stdlib. Can you merge this pull request, so that stdlib is smaller, and we continue constructive real analysis in CoRN ?

**Comment by VincentSe** [comment]
@Zimmi48 I just pushed a changelog entry for https://github.com/coq/coq/pull/12287 here.

**Review by Zimmi48 — APPROVED** [review]
LGTM. AFAICT this goes back to a state closer to what was available in 8.11, with the only significant changes being documented in the changelog.

**Comment by ejgallego** [comment]
@coq/reals-library-maintainers , it seems that it would be great to have this in 8.12, unfortunately branching is gonna happen very soon so we would need someone to process this PR. Would you mind having a look?

I took the freedom to do a rebase to get a clean CI as the two failures were spurious.

**Comment by MSoegtropIMC** [comment]
@ejgallego : the real library maintainers discussed today that this PR needs a few changes (different modularisation). I think Vincent will do it asap and I can merge over the weekend. Or das very soon mean today?

**Comment by Zimmi48** [comment]
> I think Vincent will do it asap and I can merge over the weekend

That's fine.

**Comment by VincentSe** [comment]
All constructive reals are now marked experimental.

File: doc/changelog/10-standard-library/12288-constructive-experimental.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- **Changed:**
+  Split files ConstructiveMinMax and ConstructivePower. warning::Mark all constructive reals as experimental.
```
Comment (Zimmi48) [inline doc/changelog/10-standard-library/12288-constructive-experimental.rst:2]:
```suggestion
  Split files ConstructiveMinMax and ConstructivePower.
  
  .. warning:: The constructive reals modules are marked as experimental.
```

File: doc/changelog/10-standard-library/12288-constructive-experimental.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+  (`#12287 <https://github.com/coq/coq/pull/12287>`_,
```
Comment (Zimmi48) [inline doc/changelog/10-standard-library/12288-constructive-experimental.rst:3]:
```suggestion
  (`#12288 <https://github.com/coq/coq/pull/12288>`_,
```

**Comment by VincentSe** [comment]
@MSoegtropIMC Hello, I did the cleanup we discussed. You can merge now.

**Comment by MSoegtropIMC** [comment]
@VincentSe : I restarted CI - is didn't start for some reason.

**Comment by VincentSe** [comment]
@Zimmi48 Your proposed changelog modifications do not compile. Can you fix them ?
`doc/unreleased.rst:716:Explicit markup ends without a blank line; unexpected unindent.`

**Comment by Zimmi48** [comment]
Sorry will do

**Review by MSoegtropIMC — APPROVED** [review]
LGTM. I will merge as soon as CI passes.

**Comment by Zimmi48** [comment]
Done.

**Comment by VincentSe** [comment]
@MSoegtropIMC The last error looks spurious.

**Comment by Zimmi48** [comment]
> @MSoegtropIMC The last error looks spurious.

Yes, and now someone has restarted it and it passed.

**Comment by MSoegtropIMC** [comment]
> Yes, and now someone has restarted it and it passed.

Yes, that was me. As a small kid I have seen the movie "Bernhard & Bianca" where two mice are flying with an "Albatros airline". The start checklist ends with "If it doesn't work the first time, try again". An essential movie for any user of a CI ;-)

**Comment by VincentSe** [comment]
@MSoegtropIMC It it the German version ? In French it's "Bernard & Bianca".

**Comment by MSoegtropIMC** [comment]
> It it the German version ? In French it's "Bernard & Bianca".

Yes, the German version has the same title. I learned French quite a bit later and unfortunately meanwhile forgot most of it. I need to refresh it - maybe by watching the movie in French ;-)

**Pull request merged**